evmrpc/worker_pool_metrics.go

```diff
@@ -0,0 +1,874 @@
+	DBSemaphoreWaitCount  atomic.Int64
+
+	// eth_getLogs specific stats
+	GetLogsRequests      atomic.Int64
+	GetLogsErrors        atomic.Int64
+	GetLogsSuccess       atomic.Int64 // Successful requests
+	GetLogsBlockRangeSum atomic.Int64 // Sum of block ranges for average calculation
+	GetLogsLatencySumNs  atomic.Int64 // Sum of latencies for average calculation
+	GetLogsPeakRange     atomic.Int64
+	GetLogsMaxLatencyNs  atomic.Int64 // Max latency observed
+
+	// Error type breakdown
+	ErrRangeTooLarge atomic.Int64
+	ErrTimeout       atomic.Int64
+	ErrRateLimited   atomic.Int64
+	ErrBackpressure  atomic.Int64
+	ErrIOSaturated   atomic.Int64
+	ErrBlockNotFound atomic.Int64
+	ErrQueueFull     atomic.Int64
+	ErrOther         atomic.Int64
+
+	// Block range distribution buckets (total requests)
+	RangeBucket1to10      atomic.Int64 // 1-10 blocks
+	RangeBucket11to100    atomic.Int64 // 11-100 blocks
+	RangeBucket101to500   atomic.Int64 // 101-500 blocks
+	RangeBucket501to1000  atomic.Int64 // 501-1000 blocks
+	RangeBucket1001to2000 atomic.Int64 // 1001-2000 blocks
+	RangeBucketOver2000   atomic.Int64 // >2000 blocks
+
+	// Block range success counts (for calculating success rate per bucket)
+	RangeBucket1to10Success      atomic.Int64
+	RangeBucket11to100Success    atomic.Int64
+	RangeBucket101to500Success   atomic.Int64
+	RangeBucket501to1000Success  atomic.Int64
+	RangeBucket1001to2000Success atomic.Int64
+	RangeBucketOver2000Success   atomic.Int64
+
+	// Subscription stats
+	ActiveSubscriptions atomic.Int32
+	SubscriptionErrors  atomic.Int64
+
+	// Time window for TPS calculation
+	windowStart    time.Time
+	windowRequests atomic.Int64
+	mu             sync.RWMutex
+}
+
+var (
+	metricsPrinterOnce sync.Once
+	metricsStopChan    chan struct{}
+)
+
+// GetGlobalMetrics returns the metrics from the global worker pool
+// This is a convenience function for accessing metrics without importing worker pool
+func GetGlobalMetrics() *WorkerPoolMetrics {
+	return GetGlobalWorkerPool().Metrics
+}
+
+// StartMetricsPrinter starts a background goroutine that prints metrics every interval
+// This is idempotent - only the first call will start the printer
+// Note: Printing to stdout is controlled by the EVM_DEBUG_METRICS environment variable
+// Set EVM_DEBUG_METRICS=true to enable debug output
+func StartMetricsPrinter(interval time.Duration) {
+	metricsPrinterOnce.Do(func() {
+		metricsStopChan = make(chan struct{})
+		debugEnabled := IsDebugMetricsEnabled()
+		go func() {
+			ticker := time.NewTicker(interval)
+			defer ticker.Stop()
+
+			for {
+				select {
+				case <-ticker.C:
+					m := GetGlobalMetrics()
+					// Export to Prometheus (gauges need periodic update)
+					m.ExportPrometheusMetrics()
+					// Print to stdout only if debug is enabled
+					if debugEnabled {
+						m.PrintMetrics()
+					}
+				case <-metricsStopChan:
+					return
+				}
+			}
+		}()
+	})
+}
+
+// StopMetricsPrinter stops the metrics printer
+func StopMetricsPrinter() {
```

Comment:
As far as I can see this not called from anywhere, which means there is nothing to stop the goroutine started by `StartMetricsPrinter`. 

This one is a blocker.